core/include/moveit/task_constructor/properties.h

```diff
@@ -101,18 +131,39 @@ class Property
 	/// reset to default value (which can be empty)
 	void reset();
 
-	/// the current value defined or will the fallback be used?
-	inline bool defined() const { return !value_.empty(); }
+	/// is a value defined?
+	inline bool defined() const { return !(value_.empty() && default_.empty()); }
```

Comment:
Here you change the semantics of `defined()`!
I guess you want to distinguish `isDefined()` from `isSet() == hasCurrentValue()`?
Let's discuss the naming over phone...